### PR TITLE
adding vectorstore_kwarg attribute to search_similarity function

### PR DESCRIPTION
**Pull request by amaleki2**
- **Description:** the ability to add all extra parameter of vectorstore and using them SemanticSimilarityExampleSelector.
  - **Issue:** #14583
  - **Dependencies:** no dependensies
  - **Tag maintainer:** 
  - **Twitter handle:** @AmirMalekiz

### CODE REVIEW

File: libs/core/langchain_core/example_selectors/semantic_similarity.py [hunk 1]
```diff
@@ -28,6 +28,8 @@ class SemanticSimilarityExampleSelector(BaseExampleSelector, BaseModel):
     input_keys: Optional[List[str]] = None
     """Optional keys to filter input to. If provided, the search is based on
     the input variables instead of all variables."""
+    vectorstore_kwargs: Optional[Dict[str, Any]]
```
Comment (hwchase17) [inline libs/core/langchain_core/example_selectors/semantic_similarity.py:31]: Lets make this default to None (backwards compatible)

File: libs/core/langchain_core/example_selectors/semantic_similarity.py [hunk 2]
```diff
@@ -28,6 +28,8 @@ class SemanticSimilarityExampleSelector(BaseExampleSelector, BaseModel):
...
+    """extra arguments passed to similarity_search function of the vectorstore"""
```
Comment (hwchase17) [inline libs/core/langchain_core/example_selectors/semantic_similarity.py:32]: Capitalize, and end in period

File: libs/core/langchain_core/example_selectors/semantic_similarity.py (hunk 1 above)
Comment (amaleki2) [reply libs/core/langchain_core/example_selectors/semantic_similarity.py]: ops, I missed that

**Comment by amaleki2** [comment]
@hwchase17 can you please take a second look?

**Review by hwchase17 — APPROVED** [review]
thanks!

**Pull request merged**